src/libopensc/reader-pcsc.c

```diff
@@ -585,7 +585,7 @@ static void initialize_uid(sc_reader_t *reader)
 		if (SC_SUCCESS == pcsc_transmit(reader, &apdu)
 				&& apdu.sw1 == 0x90 && apdu.sw2 == 0x00) {
 			reader->uid.len = apdu.resplen;
-			memcpy(reader->uid.value, apdu.resp, reader->uid.len);
+			memcpy(reader->uid.value, apdu.resp, MIN(sizeof(reader->uid.value), reader->uid.len));
```

Comment:
uid.len is 0 at this point, so your fix won't work. Also, there are some corner cases that aren't catched

Reply:
This is missing the adjustment of the `reader->uid.len` value if we get longer response. I like the #2232 better as it handles the issue outside of the block.